Chat/src/app/HomeScreen.tsx

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  IImageStyles,
+  Icon,
+  Image,
+  Link,
+  List,
+  PrimaryButton,
+  Spinner,
+  Stack,
+  Text,
+  mergeStyles
+} from '@fluentui/react';
+import React, { useCallback, useState } from 'react';
+import {
+  buttonStyle,
+  buttonWithIconStyles,
+  configContainerStackTokens,
+  configContainerStyle,
+  containerTokens,
+  containerStyle,
+  headerStyle,
+  listIconStyle,
+  listItemStackTokens,
+  listItemStyle,
+  imgStyle,
+  listStyle,
+  nestedStackTokens,
+  infoContainerStyle,
+  infoContainerStackTokens,
+  videoCameraIconStyle
+} from './styles/HomeScreen.styles';
+import { useTheme } from '@azure/communication-react';
+import { ThemeSelector } from './theming/ThemeSelector';
+import { useSwitchableFluentTheme } from './theming/SwitchableFluentThemeProvider';
+import { Chat20Filled } from '@fluentui/react-icons';
+import heroSVG from '../assets/hero.svg';
+import heroDarkModeSVG from '../assets/hero_dark.svg';
+import { getThreadId } from './utils/getThreadId';
+import { createThread } from './utils/createThread';
+
+const imageStyleProps: IImageStyles = {
+  image: {
+    height: '100%'
+  },
+  root: {}
+};
+
+const HOMESCREEN_SHOWING_START_CHAT_BUTTON = 1;
+const HOMESCREEN_SHOWING_LOADING_SPINNER_CREATE_THREAD = 2;
+
+/**
+ * HomeScreen has two states:
+ * 1. Showing start chat button
+ * 2. Showing spinner after clicking start chat
+ *
+ * @param props
+ */
+export default (): JSX.Element => {
+  const spinnerLabel = 'Creating a new chat thread...';
+  const iconName = 'SkypeCircleCheck';
+  const headerTitle = 'Exceptionally simple chat app';
+  const startChatButtonText = 'Start chat';
+  const listItems = [
+    'Launch a conversation with a single click',
+    'Real-time messaging with indicators',
+    'Invite up to 250 participants',
+    'Learn more about this'
+  ];
+
+  const [homeScreenState, setHomeScreenState] = useState<number>(HOMESCREEN_SHOWING_START_CHAT_BUTTON);
+  const { currentTheme } = useSwitchableFluentTheme();
+
+  const imageProps = { src: currentTheme.name === 'Light' ? heroSVG.toString() : heroDarkModeSVG.toString() };
+
+  const onCreateThread = async (): Promise<void> => {
+    const exisitedThreadId = getThreadId();
+    if (exisitedThreadId && exisitedThreadId.length > 0) {
+      setHomeScreenState(HOMESCREEN_SHOWING_LOADING_SPINNER_CREATE_THREAD);
```

Comment:
this should be outside the if statement as this always happens

Reply:
Created ADO Item
https://skype.visualstudio.com/SPOOL/_workitems/edit/2691379